widgetry/src/backend_glow.rs

```diff
@@ -467,36 +496,11 @@ pub fn load_textures(
                 sprite_height,
             );
             for p in sprite_cell.pixels() {
-                formatted_pixels.push(p.2.clone());
+                formatted_pixel_bytes.extend_from_slice(&p.2 .0);
             }
         }
     }
 
-    // Allocate the storage.
-    unsafe {
-        gl.tex_storage_3d(
-            target,
-            mipmap_level,
-            internal_format,
-            sprite_length as i32,
-            sprite_height as i32,
-            sprite_count as i32,
-        );
-    }
-
-    // Upload pixel data.
-    //
-    // From: https://www.khronos.org/opengl/wiki/Array_Texture#Creation_and_Management
-    // > The first 0 refers to the mipmap level (level 0, since there's only 1)
-    // > The following 2 zeroes refers to the x and y offsets in case you only want to
-    // > specify a subrectangle.
-    // > The final 0 refers to the layer index offset (we start from index 0 and have 2
-    // > levels).
-    // > Altogether you can specify a 3D box subset of the overall texture, but only one
-    // > mip level at a time.
-    let formatted_pixel_bytes: Vec<u8> =
-        formatted_pixels.iter().flat_map(|p| p.0.to_vec()).collect();
```

Comment:
This was the main cost - rather than building up an intermediate array of byte slices and collapsing them, we instead just write each slices to the final vec.

Reply:
A vector per pixel does seem like a bit of a problem, since I think the contents wind up on the heap. :)  Fix makes sense.